services/tenant-ui/frontend/src/components/reservation/status/Approved.vue

```diff
@@ -87,10 +105,34 @@ const handleSubmit = async (isFormValid: boolean) => {
       reservation.value.reservation_id,
       formFields.password
     );
-  } catch (error) {
-    toast.error(`Failure: ${error}`);
+  } catch (error: any) {
+    /**
+     * If error is 401, check if the reservation is expired.
+     * If not expired, show the incorrect password error.
+     * Otherwise send the error to Toast
+     */
+    const resp = error.response;
+    const exp = resp.data.match(/expired/i);
+    if (resp.status === 401 && exp) {
+      errorMessage.value = 'Reservation has expired.';
+      showError.value = true;
+    } else if (resp.status === 401) {
+      errorMessage.value = 'Incorrect password. Please try again.';
+      showError.value = true;
+    } else {
+      toast.error(resp.data);
+    }
+    console.error('message', resp.data);
+    console.error('status', resp.status);
```

Comment:
Great! Too bad to have to string check for 'expired' but no real other options in terms of API response so all good